bundles/org.eclipse.passage.lbc.equinox/src/org/eclipse/passage/lbc/internal/equinox/conditions/AcquireConditionActionExecutor.java

```diff
@@ -50,68 +48,33 @@
 @Component(property = LicensingNet.ACTION + '=' + ConditionActions.ACQUIRE)
 public class AcquireConditionActionExecutor implements BackendActionExecutor {
 
-	private static final Logger LOG = LoggerFactory.getLogger(AcquireConditionActionExecutor.class);
-
-	private List<ConditionMiner> licenseConditionMiners = new ArrayList<>();
-	private Map<String, ConditionTransport> mapCondition2Transport = new HashMap<>();
+	private final BackendLicenseVault baseLicenseVault = new BaseLicenseVault();
 
 	@Override
 	public LicensingResult executeAction(HttpServletRequest request, HttpServletResponse response) {
 		String source = getClass().getName();
-		LOG.info(NLS.bind(EquinoxMessages.AcquireConditionActionExecutor_i_execute_action, source));
-		if (licenseConditionMiners.isEmpty()) {
-			String error = EquinoxMessages.AcquireConditionActionExecutor_e_no_miners;
-			LOG.error(error);
-			return LicensingResults.createError(error, source);
-		}
 		try {
-			String productId = request.getParameter(LicensingConfigurations.LICENSING_PRODUCT_IDENTIFIER);
-			String productVersion = request.getParameter(LicensingConfigurations.LICENSING_PRODUCT_VERSION);
-			LicensingConfiguration configuration = LicensingConfigurations.create(productId, productVersion);
-
-			Collection<LicensingCondition> resultConditions = new ArrayList<>();
-
-			for (ConditionMiner miner : licenseConditionMiners) {
-				Iterable<LicensingCondition> descriptors = miner.extractLicensingConditions(configuration);
-				resultConditions.addAll((Collection<? extends LicensingCondition>) descriptors);
-			}
-			String contentType = request.getParameter(LicensingProperties.LICENSING_CONTENT_TYPE);
-			ConditionTransport transport = mapCondition2Transport.get(contentType);
-			if (transport == null) {
-				String error = NLS.bind(EquinoxMessages.AcquireConditionActionExecutor_e_transport_not_defined,
-						contentType);
-				LOG.error(error);
-				return LicensingResults.createError(error, source);
-			}
-
-			transport.writeConditions(resultConditions, response.getOutputStream());
+			Collection<ConditionPack> availableLicenses = baseLicenseVault.availableLicenses(miningRequest(request));
```

Comment:
Please extract several methods from there:
1) create and setup the Resource
2) add to Resource and save